src/modm/platform/heap/cortex/no_heap.c.in

```diff
@@ -19,7 +19,7 @@ void __modm_initialize_memory(void)
 }
 
 // ----------------------------------------------------------------------------
-modm_weak modm_section("{{ no_heap_section }}")
+modm_weak modm_used modm_section("{{ no_heap_section }}")
```

Comment:
Wouldn't that always trigger the warning of the heap module not being included even if heap is not used in the application code?

Reply:
When heap is used, no_heap.cpp is excluded from build, isn't it?

Reply:
Yeah, but if heap is not used, then this weak `_sbrk_r` acts as a "canary function" that the linker tries to link if someone tries to use malloc (which internally calls `_sbrk_r` via Newlib.
This function is normally garbage collected by the linker because nobody calls it, but is then `DISCARD`ed in the linkerscript, thus the [linker will fail with an error message](https://modm.io/reference/module/modm-platform-heap-rp--sam--stm32/#heap-is-not-implemented-error).